Project2/presentation/detail/pages/ReviewItem.swift

```diff
@@ -0,0 +1,152 @@
+            cardView.rightAnchor.constraint(equalTo: contentView.rightAnchor, constant: -10),
+            cardView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -10),
+            cardView.widthAnchor.constraint(equalToConstant: contentView.frame.width),
+            
+            posterImageView.topAnchor.constraint(equalTo: cardView.topAnchor, constant: 15),
+            posterImageView.leftAnchor.constraint(equalTo: cardView.leftAnchor, constant: 15),
+            posterImageView.widthAnchor.constraint(equalToConstant: 50),
+            posterImageView.heightAnchor.constraint(equalToConstant: 50),
+            
+            userName.topAnchor.constraint(equalTo: posterImageView.topAnchor),
+            userName.leftAnchor.constraint(equalTo: posterImageView.rightAnchor, constant: 20),
+            userName.rightAnchor.constraint(lessThanOrEqualTo: cardView.rightAnchor, constant: -15),
+            
+            
+            createdBy.topAnchor.constraint(equalTo: userName.bottomAnchor, constant: 5),
+            createdBy.leftAnchor.constraint(equalTo: posterImageView.rightAnchor, constant: 20),
+            createdBy.rightAnchor.constraint(lessThanOrEqualTo: cardView.rightAnchor, constant: -15),
+            
+            content.leftAnchor.constraint(equalTo: posterImageView.rightAnchor, constant: 20),
+            content.rightAnchor.constraint(equalTo: cardView.rightAnchor, constant: -20),
+            content.topAnchor.constraint(equalTo: createdBy.bottomAnchor, constant: 10),
+            content.bottomAnchor.constraint(equalTo: cardView.bottomAnchor, constant: -10)
+        ])
+    }
+
+    func configure(with review: Review?) {
+        content.text = review?.content ?? ""
+        let username = review?.authorDetails?.username
+        if username != nil {
+            self.userName.text = username ?? ""
+        } else if review?.authorDetails?.name != nil {
+            self.userName.text = review?.authorDetails?.name ?? ""
+        }
+        createdBy.text = "Written by \(review?.authorDetails?.username ?? "") on \(formatDate(review?.createdAt) ?? "")"
+        setImage(path: review?.authorDetails?.avatarPath)
+    }
+    
+    func setImage(path:String?){
+        let image = UIImage(named: "default-avatar-icon-of-social-media-user-vector")
+        if let avatarPath = path {
+            let modifiedAvatarPath = String(avatarPath.dropFirst())
+            if let url = URL(string: modifiedAvatarPath) {
+                posterImageView.sd_setImage(with: url, placeholderImage: image)
+            } else {
+                posterImageView.image = image
+            }
+        } else {
+            posterImageView.image = image
+        }
+    }
+    
+    func formatDate(_ dateString: String?) -> String? {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"
+        
+        if dateString != nil {
```

Comment:
Optional.map(_:) ფუნქცია შეგიძლია გამოიყენო აქ

`return dateString.map { string in ... }` ეს მეპი მუშაობს ასე - თუ nil არის, მაშინ არ შემოვა ბლოკში და დააბრუნებს nil-ს, ხოლო თუ არ არის, ბლოკში შემოვა და იმას დააბრუნებს რასაც შიგნით ჩაწერ